### PR TITLE
Add shared support for MongoDB Dev Services container

### PR DESCRIPTION
**Pull request by patrox**

### CODE REVIEW

File: extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesMongoProcessor.java
Comment (gsmet) [inline extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesMongoProcessor.java:63]: The constant should probably be named like a constant.

File: extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesBuildTimeConfig.java
Comment (gsmet) [inline extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesBuildTimeConfig.java:49]: Could you make sure you use `MongoDB` instead of `Mongo` in the javadoc of the configuration properties? They are published on the website.
(Note that other properties might not comply but let's not add more :))

File: extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesMongoProcessor.java
Comment (gsmet) [inline extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesMongoProcessor.java:61]:
```suggestion
    private static final String DEV_SERVICE_LABEL = "quarkus-dev-service-mongodb";
```

File: docs/src/main/asciidoc/mongodb-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-dev-services.adoc:27]: Any chance we could make it `mongodb` or it is a convention and we should keep it that way?

File: docs/src/main/asciidoc/mongodb-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-dev-services.adoc:23]:
```suggestion
NOTE: Dev Services for MongoDB starts the container with the `quarkus-dev-service-mongodb` label which is used to identify the container.
```

File: docs/src/main/asciidoc/mongodb-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-dev-services.adoc:25]: I checked and I think the root config is `quarkus.mongodb` already?
```suggestion
If you need multiple (shared) servers, you can configure the `quarkus.mongodb.devservices.service-name` attribute and indicate the server name.
```

File: docs/src/main/asciidoc/mongodb-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-dev-services.adoc:6]: Let's not clutter our guide index:
```suggestion
= Dev Services for MongoDB
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks for this PR. It looks very nice.

I added a few minor gripes inline as to preferring `MongoDB`/`mongodb` over mongo. I'm not sure it makes sense in all circumstances so feel free to push back if it doesn't.

Could you also squash all your commits when done?

Thanks!

File: extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesBuildTimeConfig.java
Comment (patrox) [reply extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesBuildTimeConfig.java:49]: Sure, will fix it. Naming is so hard ;)

File: docs/src/main/asciidoc/mongodb-dev-services.adoc
Comment (patrox) [reply docs/src/main/asciidoc/mongodb-dev-services.adoc:27]: It can be changed

**Comment by patrox** [comment]
> Thanks for this PR. It looks very nice.
> 
> I added a few minor gripes inline as to preferring `MongoDB`/`mongodb` over mongo. I'm not sure it makes sense in all circumstances so feel free to push back if it doesn't.
> 
> Could you also squash all your commits when done?
> 
> Thanks!

@gsmet I've addressed all your suggestions/comments and squashed all commits after rebasing to the current `main`.

File: docs/src/main/asciidoc/mongodb-dev-services.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/mongodb-dev-services.adoc:27]: @patrox are you sure you pushed your changes? I don't see my comments addressed and also you said you squashed everything and the commits are not.

Thanks for your feedback!

Comment (patrox) [reply docs/src/main/asciidoc/mongodb-dev-services.adoc]: > @patrox are you sure you pushed your changes? I don't see my comments addressed and also you said you squashed everything and the commits are not.
> 
> Thanks for your feedback!

Yeah sorry, it looks like I messed things up during rebase. Now it should be fine :)

Comment (patrox) [reply docs/src/main/asciidoc/mongodb-dev-services.adoc]: @gsmet Now all requested changes should be in place, the commits were squashed and the branch is up-to-date with the current `main`, so it looks ready!

**Review by gsmet — APPROVED** [review]
I amended your commit to fix `DEV_SERVICE_LABEL` that was still the `mongo` version.

LGTM, let's merge when CI is green.

Sorry it took so much time for me to get back to it :).

**Comment by patrox** [comment]
> I amended your commit to fix `DEV_SERVICE_LABEL` that was still the `mongo` version.
> 
> LGTM, let's merge when CI is green.
> 
> Sorry it took so much time for me to get back to it :).

Thank you @gsmet, will wait for the CI to become 🟢

**Comment by gsmet** [comment]
@patrox thanks for your contribution, it's going to be part of 3.12.

**Pull request merged**
